### ISSUE TITLE
New cop to detect unsafe comparison method definitions

### ISSUE THREAD

**Issue by fatkodima** [rubocop/rubocop#12289]
When defining comparison methods (`==` and `eql?`), it is easy to miss that the `other` should be of the same class (or some other class which we should also check explicitly) and write a non safe method.

Example: 
```ruby
# bad
def ==(other)
  attributes == other.attributes
end

# good
def ==(other)
  self.class == other.class && # or kind_of? or is_a? or instance_of?
    attributes == other.attributes
end

# also good
def ==(other)
  attributes == other&.attributes
end
```

Inspired by https://github.com/rails/rails/issues/49670.

**Comment by viralpraxis** [comment]
@fatkodima is your third example safe? I'd expect `==` not to raise if `other` does not respond to `attributes` message

**Comment by fatkodima** [comment]
Thanks, correct, removed that example. It was basically the same as a "bad" example.
